src/agent/nodes/_share.py

```
@@ -38,62 +340,267 @@ def invoke_with_self_repair(
         errors = local_config.validate_references()
 
         if not errors:
             if attempt > 0:
-                logger.info("[{}] self-repair succeeded on round {}", phase, attempt)
+                logger.info(
+                    "[{}] self-repair succeeded on round {}", phase, attempt
+                )
             return result
 
         if attempt == MAX_SELF_REPAIR_ROUNDS:
+            if gap:
```

Ruff (0.16.1)

[error] Undefined name `gap`

(F821)